src/walktree.js

```diff
@@ -1,49 +1,60 @@
-const fs = require('fs');
+var fs = require('fs');
+var path = require('path');
 
-const walkTree = function walkTree(opts) {
-    const path = opts.root;
-    const ignored = opts.ignore;
-    const create = opts.create;
-    const countFolders = opts.countFoldersAsFiles;
-    const fileName = opts.file;
 
-    walkTreeRecursive(path);
+function walkTree (opts) {
+    var create = opts.create;
+    var ignored = opts.ignore;
+    var countFolders = opts.countFoldersAsFiles;
+    var nameOfPlaceholder = opts.file;
 
-    function walkTreeRecursive(path) {
-        let elements = fs.readdirSync(path);
-        let fileCount = 0;
+    walkTreeRecursive(opts.root);
 
-        while (elements.length > 0) {
-            let element = elements.pop();
-            let elementPath = path + '/' + element;
-
-            // Skip ignored folder and tries next folder in line
-            if (ignored.includes(element))
-                continue;
+    function walkTreeRecursive(dirPath) {
+        fs.readdir(dirPath, function(err, elements) {
+            if (err) {
+              throw new Error('Failed to read the directory at: ' + dirPath);
+            }
 
-            let stat = fs.statSync(elementPath);
-            if (stat.isDirectory()) {
-                if (create && countFolders)
-                    fileCount++;
+            var fileCount = 0;
 
-                walkTreeRecursive(elementPath);
+            elements.forEach(function(element) {
+                var fullPathToElement = path.join(dirPath, element);
 
-            } else if (stat.isFile()) {
-                fileCount++;
+                // Skip ignored folder and tries next folder in line
+                if (ignored.includes(element) || !fs.existsSync(fullPathToElement)) {
+                    return;
+                }
 
-                if (!create && element === fileName)
-                    fs.unlinkSync(elementPath);
+                var stat = fs.statSync(fullPathToElement);
+                if (stat.isDirectory()) {
+                    if (create && countFolders) {
+                        fileCount++;
+                    }
 
-            } else {
-                return;
-            }
-        } // end of While loop
+                    walkTreeRecursive(fullPathToElement);
 
-        // Creates a file
-        if (create && fileCount === 0)
-            fs.closeSync(fs.openSync(path + '/' + fileName, 'w'));
+                } else if (stat.isFile()) {
+                    fileCount++;
 
-        return;
+                    if (!create && element === nameOfPlaceholder) {
+                        fs.unlink(fullPathToElement);
+                    }
+
+                }
+            });
+
+            // Creates a file
+            if (create && fileCount === 0) {
+                var pathToPlaceholder = path.join(dirPath, nameOfPlaceholder);
```

Comment:
You'll find that I tweaked variable names in certain places to make them more clear, especially because here we are dealing with potentially four or five different variables that have to do with "files" or "directories" per scope. Don't be afraid of making longish variable names that are semantic - it helps code readability :)